torch/nn/utils/parametrize.py

```diff
@@ -341,9 +341,17 @@ def remove_parametrizations(
         )
 
     # Fetch the original tensor
-    original = module.parametrizations[tensor_name].original  # type: ignore[index, union-attr]
+    parametrizations = module.parametrizations[tensor_name]  # type: ignore[index, union-attr]
+    original = parametrizations.original  # type: ignore[union-attr]
     if leave_parametrized:
-        t = getattr(module, tensor_name)
+        # Call getattr in eval mode so that parametrizations that
+        # compute weight iteratively won't perform another iteration
+        training = parametrizations.training   # type: ignore[union-attr]
+        parametrizations.eval()  # type: ignore[union-attr]
```

Comment:
That looks like a pretty big change and that can have pretty big impact. I don't think you want to change that.

Reply:
Hmm, aren't the parametrizations about to be removed? We also restore it to its original state since the user may be holding a pointer as @Lezcano mentioned. 

One issue I see with the current PR is that individual parametrizations may have different values of eval vs train. If that is the only issue, we could save and restore those values individually over a loop

Another way is to deepcopy the parametrizations and invoke its forward directly instead of doing getattr

Reply:
What I mean is that this can be surprising for the user that the removal of the reparametrization happens in eval mode even though they didn't ask for it.

I would feel better without this code and a Note in the spectral_norm doc: you should put the Module in eval() mode before removing the reparametrization if you don't want to do an extra iteration.

Reply:
Removed the code and added the note.
Side note:  if we register a parametrization on a module in eval mode, its parametrization module would actually still be training mode, and this may cause it to perform updates silently. To be consistent with the behavior of the old
spectral norm, maybe newly registered parametrizations should default to the training mode of its host module.

```
pre_remove_out = m(input)
m.eval() # this works as expected because SpectralNorm is a submodule
m = torch.nn.utils.parametrize.remove_parametrizations(m, 'weight')
self.assertEqual(m(input), pre_remove_out)

torch.nn.utils.parametrizations.spectral_norm(m)
pre_remove_out = m(input)
self.assertTrue(m.parametrizations.weight[0].training) # m is actually still in eval mode, but newly registered
                                                                                     # parametrization is in training mode
m = torch.nn.utils.parametrize.remove_parametrizations (m, 'weight')
self.assertNotEqual(m(input), pre_remove_out)
```

Reply:
training mode of its host Module sounds good to me, what do you think @Lezcano

Reply:
Update: I've tentatively added this feature in this PR

Reply:
I think training mode of the host is reasonable.